src/test/librbd/mirror/snapshot/test_mock_CreatePrimaryRequest.cc

```diff
@@ -181,8 +184,10 @@ class TestMockMirrorSnapshotCreatePrimaryRequest : public TestMockFixture {
                              boost::get<cls::rbd::MirrorSnapshotNamespace>(
                                &it->second.snap_namespace);
                            ASSERT_NE(nullptr, info);
-                           ASSERT_NE(0, info->mirror_peer_uuids.erase(
-                                       peer_uuid));
+                           if (!no_peer_test) {
+                             ASSERT_NE(0, info->mirror_peer_uuids.erase(
+                                         peer_uuid));
+                           }
```

Comment:
Instead of `no_peer_test`, I'd pass whether the peer is expected to be in the set and `ASSERT_EQ` on that unconditionally.